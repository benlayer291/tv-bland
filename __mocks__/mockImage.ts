import { InterfaceTvShowImage } from '../types/interfaces'

export const mockImage: InterfaceTvShowImage = {
  medium:
    'http://static.tvmaze.com/uploads/images/medium_portrait/81/202627.jpg',
  original:
    'http://static.tvmaze.com/uploads/images/original_untouched/81/202627.jpg',
}
