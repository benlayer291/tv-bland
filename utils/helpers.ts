export const removeProtocolFromUrl = (url: string): string =>
  url.replace(/^https?:\/\//i, '//')

export const removeHtmlTagsFromString = (htmlString: string): string =>
  htmlString.replace(/(<([^>]+)>)/gi, '')

export const checkDeviceIsMobile = (): boolean => {
  return (
    typeof navigator !== 'undefined' &&
    /Android|Mobi/i.test(navigator.userAgent)
  )
}
