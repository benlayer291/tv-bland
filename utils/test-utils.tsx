import { render, RenderOptions } from '@testing-library/react'

type Props = {
  children: React.ReactElement
}

const Providers: React.FunctionComponent<Props> = ({ children }: Props) => {
  return <>{children}</>
}

const customRender = (
  ui: React.ReactElement,
  options: Omit<RenderOptions, 'queries'> = {}
) => render(ui, { wrapper: Providers, ...options })

// re-export everything
export * from '@testing-library/react'

// override render method
export { customRender as render }
