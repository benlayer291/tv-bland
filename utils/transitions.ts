type Transtion = {
  duration: number
  ease: number[]
}

export const transition: Transtion = {
  duration: 5,
  ease: [0.43, 0.13, 0.23, 0.96],
}
