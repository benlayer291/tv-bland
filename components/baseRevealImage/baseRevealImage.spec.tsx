import React from 'react'
import { act, render, RenderResult } from '../../utils/test-utils'

import { mockImage } from '../../__mocks__/mockImage'

import BaseRevealImage from './baseRevealImage'

const propsDefault = {
  src: mockImage.medium,
  alt: 'Alt text',
  height: 295,
  width: 210,
}

jest.useFakeTimers()

describe('<BaseRevealImage />', () => {
  let el: RenderResult

  it('should render', () => {
    el = render(<BaseRevealImage {...propsDefault} />)

    expect(el.container).toBeInTheDocument()
  })

  it('should match snapshot', () => {
    el = render(<BaseRevealImage {...propsDefault} />)

    expect(el.container).toMatchSnapshot()
  })

  it('should reveal image after time delay', () => {
    el = render(<BaseRevealImage {...propsDefault} />)

    act(() => jest.advanceTimersByTime(10000))

    expect(el.getByTestId('base-reveal-image')).toHaveClass(
      'BaseRevealImage--revealed'
    )
  })

  describe('given no image src is passed', () => {
    it('should render', () => {
      el = render(<BaseRevealImage {...propsDefault} src={undefined} />)

      expect(el.container).toBeInTheDocument()
    })

    it('it should not contain image', () => {
      el = render(<BaseRevealImage {...propsDefault} src={undefined} />)

      expect(el.queryByRole('image')).not.toBeInTheDocument()
    })

    it('it should display fallback', () => {
      el = render(<BaseRevealImage {...propsDefault} src={undefined} />)

      expect(el.getByTestId('base-reveal-image')).toHaveClass(
        'BaseRevealImage--no-image'
      )
    })
  })
})
