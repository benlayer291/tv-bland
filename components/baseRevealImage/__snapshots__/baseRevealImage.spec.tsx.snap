// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BaseRevealImage /> should match snapshot 1`] = `
<div>
  <div
    class="BaseRevealImage"
    data-testid="base-reveal-image"
  >
    <img
      alt="Alt text"
      class="BaseRevealImage__image"
      height="295"
      loading="lazy"
      src="//static.tvmaze.com/uploads/images/medium_portrait/81/202627.jpg"
      width="210"
    />
    <img
      alt="Alt text"
      class="BaseRevealImage__filter"
      height="295"
      loading="lazy"
      src="//static.tvmaze.com/uploads/images/medium_portrait/81/202627.jpg"
      width="210"
    />
  </div>
</div>
`;
